app/assets/javascripts/components/article_finder/article_finder.jsx

```diff
@@ -52,20 +55,40 @@ const ArticleFinder = createReactClass({
     }
   },
 
+  getParamsURL() {
+    const query = qs.parse(window.location.search);
+    const entries = Object.entries(query);
+    entries.map(([key, val]) => {
+      val = (key === 'article_quality') ? parseInt(val) : val;
+      return this.updateFields(key, val);
+    });
+  },
   updateFields(key, value) {
-    return this.props.updateFields(key, value);
+    const update_field = this.props.updateFields(key, value);
+    Promise.resolve(update_field).then(() => {
+    if (this.props.search_term.length !== 0) { this.buildURL(); }
+    });
   },
 
   toggleFilter() {
     return this.setState({
       showFilters: !this.state.showFilters
     });
   },
-
+  buildURL() {
+    let queryStringUrl = window.location.href.split('?')[0];
```

Comment:
Should we use `window.location.search` here as well?

Reply:
window.location.search gives results after the "?" , it's better to use window.location.href as it gives the full address , that's why didn't changed it.

Reference : https://webplatform.github.io/docs/apis/location/search/

Reply:
Ah, great!  :)

Reply:
Done! :D